openfe/protocols/openmm_md/plain_md_methods.py

```diff
@@ -0,0 +1,464 @@
+            thermo_settings=thermo_settings,
+            system_settings=system_settings,
+            cache=ffcache,
+            has_solvent=solvent_comp is not None,
+        )
+
+        # b. force the creation of parameters
+        # This is necessary because we need to have the FF generated ahead of
+        # solvating the system.
+        # Note: by default this is cached to ctx.shared/db.json so shouldn't
+        # incur too large a cost
+        for comp in small_mols:
+            offmol = comp.to_openff()
+            system_generator.create_system(offmol.to_topology().to_openmm(),
+                                           molecules=[offmol])
+
+        # c. get OpenMM Modeller + a dictionary of resids for each component
+        stateA_modeller, comp_resids = system_creation.get_omm_modeller(
+            protein_comp=protein_comp,
+            solvent_comp=solvent_comp,
+            small_mols=small_mols,
+            omm_forcefield=system_generator.forcefield,
+            solvent_settings=solvation_settings,
+        )
+
+        # d. get topology & positions
+        # Note: roundtrip positions to remove vec3 issues
+        stateA_topology = stateA_modeller.getTopology()
+        stateA_positions = to_openmm(
+            from_openmm(stateA_modeller.getPositions())
+        )
+
+        # e. create the stateA System
+        stateA_system = system_generator.create_system(
+            stateA_topology,
+            molecules=[s.to_openff() for s in small_mols],
+        )
+
+        # 10. Get platform
+        platform = compute.get_openmm_platform(
+            protocol_settings.engine_settings.compute_platform
+        )
+
+        # 11. Set the integrator
+        integrator = openmm.LangevinMiddleIntegrator(
+            to_openmm(thermo_settings.temperature),
+            to_openmm(integrator_settings.collision_rate),
+            to_openmm(integrator_settings.timestep),
+        )
+
+        simulation = openmm.app.Simulation(
+            stateA_modeller.topology,
+            stateA_system,
+            integrator,
+            platform=platform
+        )
+
+        try:
+
+            if not dry:  # pragma: no-cover
+
+                simulation.context.setPositions(stateA_positions)
+
+                # minimize
+                if verbose:
+                    logger.info("minimizing systems")
+
+                simulation.minimizeEnergy(
+                    maxIterations=sim_settings.minimization_steps
+                )
+                min_pdb_file = "minimized.pdb"
+                positions = simulation.context.getState(
+                    getPositions=True, enforcePeriodicBox=False
+                ).getPositions()
+                with open(shared_basepath / min_pdb_file, "w") as f:
+                    openmm.app.PDBFile.writeFile(
+                        simulation.topology, positions, file=f, keepIds=True
+                    )
+                # equilibrate
```

Comment:
Should we set the reporter early and write out an equilibration file too? Or maybe just do a single MD step and let folks filter out their equilibration themselves?